Update documentation to note dangers of shared state in `random` module (#93743)
python/cpython: Doc/library/random.rst

for cryptographic purposes.

The functions supplied by this module are actually bound methods of a hidden

> instance of the :class:`random.Random` class, and thus share state – in 

Reviewer: ```suggestion
instance of the :class:`Random` class, and thus share state – in 
```

particular, calls to :meth:`random.set_seed` anywhere in application code 
or any libraries will affect the sequence of random values produced by calls 

> to `random.*` methods in the same process, which can lead to unexpected results.

Reviewer: ```suggestion
to ``random.*`` methods in the same process, which can lead to unexpected results.
```

or 

"other top-level functions in :mod:\`random`"

> particular, calls to :meth:`random.set_seed` anywhere in application code 
> or any libraries will affect the sequence of random values produced by calls 
> to `random.*` methods in the same process, which can lead to unexpected results.

Reviewer: Not reflowed; See my other suggestion on "random.*"

```suggestion
particular, a call to :meth:`random.set_seed` in any executed code 
will affect the sequence of random values produced, as the random seed is shared.
This affects return values from calls to other `random.*` methods in the same process, potentially leading to unexpected results.
```

> instance of the :class:`random.Random` class, and thus share state – in 
> particular, calls to :meth:`random.set_seed` anywhere in application code 
> or any libraries will affect the sequence of random values produced by calls 
> to `random.*` methods in the same process, which can lead to unexpected results.
> You should consider instantiating your own instance or instances of :class:`Random` 
> to get generators that don't share state. 

Reviewer 2: Trailing whitespace is making the CI check fail:

```suggestion
instance of the :class:`random.Random` class, and thus share state – in
particular, calls to :meth:`random.set_seed` anywhere in application code
or any libraries will affect the sequence of random values produced by calls
to `random.*` methods in the same process, which can lead to unexpected results.
You should consider instantiating your own instance or instances of :class:`Random`
to get generators that don't share state.
```

> to get generators that don't share state. 

Reviewer: For some applications, the sharing of state might not matter -- perhaps provide a heuristic for where I'd want to instantiate my own `Random` class?
